### PR TITLE
Client login service

### PR DESCRIPTION
**Pull request by noodlemoodle**
move signin signup logic to client login service

### Pulled into https://github.com/kgrgreer/foam3/pull/3579

### CODE REVIEW

File: src/foam/nanos/controller/ApplicationController.js [hunk 1]
```diff
@@ -423,6 +425,25 @@ foam.CLASS({
       class: 'Boolean',
       name: 'initSubject'
     },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.nanos.auth.login.ClientLoginService',
+      name: 'loginService',
+      factory: function() {
+        return foam.nanos.auth.login.ClientLoginService.create({}, this.__subContext__);
```
Comment (sarthak-marwaha) [inline src/foam/nanos/controller/ApplicationController.js]: This can create itself when the client loads if you add it to services jrl

Comment (sarthak-marwaha) [reply src/foam/nanos/controller/ApplicationController.js]: Look at cssTokenOverrideService

Comment (noodlemoodle) [reply src/foam/nanos/controller/ApplicationController.js]: updated, thanks @sarthak-marwaha

**Review by jlhughes — COMMENTED** [review]
LGTM

**Review by sarthak-marwaha — COMMENTED** [review]
LGTM, should make a ticket to remove all the old SignIn/SignUp models once this is tested

**Comment by noodlemoodle** [comment]
> LGTM, should make a ticket to remove all the old SignIn/SignUp models once this is tested

https://nanopay.atlassian.net/browse/NP-9554

File: src/foam/nanos/auth/login/SignIn.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  name: 'SignIn',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'identifier',
+      required: true,
+      validationTextVisible: false,
+      label: 'Email or Username',
+      preSet: function(_, n) {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/login/SignIn.js]: Can't you just say trim: true ?

File: src/foam/nanos/auth/login/SignUp.js [hunk 3]
```diff
@@ -42,13 +48,16 @@ foam.CLASS({
...
           icon: 'images/checkmark-small-green.svg',
           onKey: true,
+          isAvailable$: X.data.emailAvailable$,
           inputValidation: /\S+@\S+\.\S+/,
           restrictedCharacters: /^[^\s]$/,
           displayMode: foam.u2.DisplayMode.RW
         };
       },
-      validateObj: function(email) {
+      validateObj: function(email, emailAvailable) {
```
Comment (sarthak-marwaha) [inline src/foam/nanos/auth/login/SignUp.js]: Can we make these validationPredicates instead? For all of them?

File: src/foam/nanos/auth/login/ClientLoginService.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                showAction: true
+              }
+            }]
+          }
+        }, this));
+      }
+    },
+    {
+      name: 'wizardVerifyEmail',
+      code: async function(x, email, username, password) {
```
Comment (sarthak-marwaha) [inline src/foam/nanos/auth/login/ClientLoginService.js:148]: Should we just use this verifyEmail flow across the board? Dont see why we need to maintain two of them

**Comment by jlhughes** [comment]
Pulled into https://github.com/kgrgreer/foam3/pull/3579

**Pull request closed without merging**